src/graphics/text.rs

```diff
@@ -345,22 +369,17 @@ impl Text {
     }
 
     /// Returns the width and height of the formatted and wrapped text.
-    pub fn dimensions(&self, context: &mut Context) -> (u32, u32) {
-        if let Ok(metrics) = self.cached_metrics.try_borrow() {
-            if let (Some(width), Some(height)) = (metrics.width, metrics.height) {
-                return (width, height);
-            }
-        }
-        self.calculate_dimensions(context)
+    pub fn dimensions(&self, context: &Context) -> (u32, u32) {
+        self.calculate_dimensions(&mut context.gfx_context.glyph_brush.borrow_mut())
     }
 
     /// Returns the width of formatted and wrapped text, in screen coordinates.
-    pub fn width(&self, context: &mut Context) -> u32 {
+    pub fn width(&self, context: &Context) -> u32 {
         self.dimensions(context).0
     }
 
     /// Returns the height of formatted and wrapped text, in screen coordinates.
-    pub fn height(&self, context: &mut Context) -> u32 {
+    pub fn height(&self, context: &Context) -> u32 {
```

Comment:
We could keep taking a mutable reference in these methods... It's not necessary anymore though, and I don't think this qualifies as a breaking change (?).